src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts

```diff
@@ -0,0 +1,41 @@
+import { NGSShelf, OfferSliceWithNGS, OfferWithNGS } from ".";
+
+export type NGSOfferRow = Record<"slice", OfferSliceWithNGS> &
+  Record<NGSShelf, OfferWithNGS | null>;
+
+const getSliceKey = (slice: OfferSliceWithNGS): string => {
+  const firstSegment = slice.segments[0];
+  const lastSegment = slice.segments[slice.segments.length - 1];
+  // TODO: Confirm whether this is the correct and complete list of comparison fields
+  return `${firstSegment.marketing_carrier.id}-${firstSegment.departing_at}-${lastSegment.arriving_at}`;
+};
+
+export const groupOffersForNGSView = (
+  offers: OfferWithNGS[],
+  sliceIndex: number
+): NGSOfferRow[] => {
+  const offersMap: Record<string, NGSOfferRow> = {};
+  offers.forEach((offer) => {
+    if (sliceIndex > offer.slices.length) {
+      throw new Error("Invalid slice index");
```

Comment:
For more clarity on the error, how about: 
```suggestion
      throw new Error("Attempted to call `groupOffersForNGSView` with an Invalid slice index");
```